### PR TITLE
Allow users to configure NTP servers

### PR DESCRIPTION
**Pull request by abhinavmpandey08**
Signed-off-by: Abhinav Pandey <abhinavmpandey08@gmail.com>

*Issue #, if available:*
https://github.com/aws/eks-anywhere/issues/4760

*Description of changes:*
Users can now configure NTP servers on their EKS-A nodes by passing time servers to the `VSphereMachineConfig` object
```yaml
kind: VSphereMachineConfig
metadata:
  name: test-etcd
spec:
  hostOSConfiguration:
    ntpConfiguration:
      servers:
        - time-a.eks-a.aws
        - time-b.eks-a.aws
        - 192.168.0.10
  .
  .
  .
```

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: pkg/api/v1alpha1/vspheremachineconfig_types.go [hunk 1]
```diff
@@ -6,18 +6,29 @@ import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
...
+}
+
+// HostOSConfiguration defines the configuration settings on the host OS.
+type HostOSConfiguration struct {
+	NTPConfiguration *NTPConfiguration `json:"ntpConfiguration"`
+}
+
+// NTPConfiguration defines the NTP configuration on the host OS.
+type NTPConfiguration struct {
+	Servers []string `json:"servers"`
```
Comment (chrisdoherty4) [inline pkg/api/v1alpha1/vspheremachineconfig_types.go:31]: Do we expect this to expand to other settings? What kind of settings could go here?

Comment (abhinavmpandey08) [reply pkg/api/v1alpha1/vspheremachineconfig_types.go:31]: Within NTP itself? I don't think we are gonna add any other config here

File: pkg/providers/vsphere/vsphere_test.go [hunk 2]
```diff
@@ -3353,3 +3353,40 @@ func TestVsphereProviderMachineConfigsManagedCluster(t *testing.T) {
 		}
 	}
 }
+
+func TestProviderGenerateDeploymentFileForBottleRocketWithNTPConfig(t *testing.T) {
```
Comment (vignesh-goutham) [inline pkg/providers/vsphere/vsphere_test.go:3357]: Can we add another test for ubuntu or redhat?

**Review by vignesh-goutham — COMMENTED** [review]
Looks good!

File: pkg/providers/vsphere/vsphere_test.go (hunk 2 above)
Comment (abhinavmpandey08) [reply pkg/providers/vsphere/vsphere_test.go:3357]: added

**Review by vignesh-goutham — APPROVED** [review]

**Pull request merged**